pkg/issuer/acme/http/internal/istio/istio.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// By utilising this file, no dependance on `istio.io/client-go` is
+// required. Used in combination with a dynamic client, it is possible
+// to create a client for these CRDs without adding a client-go dependency.
+
+package istio
+
+import (
+	v1alpha1 "istio.io/api/meta/v1alpha1"
+	networkingv1beta1 "istio.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	virtualServiceGvk = schema.GroupVersionKind{Group: "networking.istio.io", Version: "v1beta1", Kind: "VirtualService"}
+	virtualServiceGvr = schema.GroupVersionResource{Group: "networking.istio.io", Version: "v1beta1", Resource: "virtualservices"}
+)
+
+func VirtualServiceGvr() schema.GroupVersionResource {
+	return virtualServiceGvr
+}
+
+type VirtualService struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the implementation of this definition.
+	Spec networkingv1beta1.VirtualService `json:"spec,omitempty"`
```

Comment:
I think https://github.com/istio/client-go uses networkingalpha3.VirtualService here. 
I'm not sure if that matters.

Reply:
I think v1beta1 should be used, this is what I found on the internet:
https://discuss.istio.io/t/is-api-version-networking-istio-io-v1alpha3-considered-to-be-deprecated-on-1-5-0/7278

Reply:
Yes, you're right. I'd been looking at the wrong version of the Kubernetes API.